server/gc/src/main/java/org/apache/accumulo/gc/GCRun.java

```diff
@@ -160,7 +161,12 @@ public Stream<Reference> getReferences() {
     // there is a lot going on in this "one line" so see below for more info
     var tabletReferences = tabletStream.flatMap(tm -> {
       // combine all the entries read from file and scan columns in the metadata table
-      var fileStream = Stream.concat(tm.getFiles().stream(), tm.getScans().stream());
+      Stream<StoredTabletFile> fileStream = tm.getFiles().stream();
+      if (!tm.getScans().isEmpty()) {
+        // scans are normally empty so only introduce a layer of indirection when it actually
+        // exists.
+        fileStream = Stream.concat(fileStream, tm.getScans().stream());
+      }
```

Comment:
Not sure if `tm.getScans()` can change while in here, but using a variable instead of calling that twice looks a little cleaner. Also moved the comment, so it doesn't wrap

```suggestion
      // scans are normally empty, so only introduce a layer of indirection when it actually exists
      final var tmScans = tm.getScans();
      if (!tmScans.isEmpty()) {
        fileStream = Stream.concat(fileStream, tmScans.stream());
      }
```